### PR TITLE
Use pyproject.toml,  remove extra files.

### LINKED ISSUE #111: Fix dependencies in setup.py and requirements.txt
At the moment we are not able to use the new `main_oo.py` if we use `setup.py`. The `install_requires` section is out of sync of `requirements.txt`.

I think we could

- drop the `requirements.txt` file and use only `setup.py`, or
- parse `requirements.txt` in `setup.py`

Either of these, or another approach, to avoid this happening again, and fixing the issue.

### PR DESCRIPTION
**Pull request by kinow**
closes #111 

The `setup.py` file can be simplified down to just a `setup()` call (as long as we use a recent `pip` module). We can use the installation groups, so instead of `test-requirements.txt`, `mypy-requirements.txt`, etc, we can group the dependencies in a single file, within groups, and then call `pip install wdl2cwl[test]`, `pip install wdl2cwl[mypy]`, etc.

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -1,18 +1,84 @@
...
+    cwl-utils >= 0.12
+    miniwdl
+    regex
+
+[options.packages.find]
+include = wdl2cwl.*
+
+[options.extras_require]
+mypy =
+    mypy==0.941
```
[the remark this answers is not in the document]
Comment (kinow) [reply setup.cfg:54]: I wanted to say yes, or probably, but there's only one way to confirm :neckbeard: 

- Found their code to parse `setup.py` [^1]
- Cloned that repo, created a new `venv`, and activated
- Installed deps from `./dependabot-core/python/helpers/requirements.txt`
- Started `python` and ran this code:

```python
Python 3.9.12 | packaged by conda-forge | (main, Mar 24 2022, 23:25:59) 
[GCC 10.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from lib import parser
>>> parser.parse_setup('/home/kinow/Development/python/workspace/wdl-cwl-translator/')
'{"result": [{"name": "cwl-utils", "version": null, "markers": "None", "file": "setup.cfg", "requirement": ">=0.12", "requirement_type": "install_requires", "extras": []}, {"name": "miniwdl", "version": null, "markers": "None", "file": "setup.cfg", "requirement": null, "requirement_type": "install_requires", "extras": []}, {"name": "regex", "version": null, "markers": "None", "file": "setup.cfg", "requirement": null, "requirement_type": "install_requires", "extras": []}, {"name": "mypy", "version": "0.941", "markers": "None", "file": "setup.cfg", "requirement": "==0.941", "requirement_type": "extras_require:mypy", "extras": []}, {"name": "types-requests", "version": null, "markers": "None", "file": "setup.cfg", "requirement": null, "requirement_type": "extras_require:mypy", "extras": []}, {"name": "types-setuptools", "version": null, "markers": "None", "file": "setup.cfg", "requirement": null, "requirement_type": "extras_require:mypy", "extras": []}, {"name": "cwltest", "version": null, "markers": "None", "file": "setup.cfg", "requirement": null, "requirement_type": "extras_require:test", "extras": []}, {"name": "cwltool", "version": null, "markers": "None", "file": "setup.cfg", "requirement": ">=3.1.20220202110738", "requirement_type": "extras_require:test", "extras": []}, {"name": "pytest", "version": null, "markers": "None", "file": "setup.cfg", "requirement": "<7.2,>=6.0.2", "requirement_type": "extras_require:test", "extras": []}, {"name": "pytest-cov", "version": null, "markers": "None", "file": "setup.cfg", "requirement": null, "requirement_type": "extras_require:test", "extras": []}, {"name": "pytest-xdist", "version": null, "markers": "None", "file": "setup.cfg", "requirement": null, "requirement_type": "extras_require:test", "extras": []}, {"name": "mypy", "version": "0.941", "markers": "None", "file": "setup.cfg", "requirement": "==0.941", "requirement_type": "extras_require:all", "extras": []}, {"name": "types-requests", "version": null, "markers": "None", "file": "setup.cfg", "requirement": null, "requirement_type": "extras_require:all", "extras": []}, {"name": "types-setuptools", "version": null, "markers": "None", "file": "setup.cfg", "requirement": null, "requirement_type": "extras_require:all", "extras": []}, {"name": "cwltest", "version": null, "markers": "None", "file": "setup.cfg", "requirement": null, "requirement_type": "extras_require:all", "extras": []}, {"name": "cwltool", "version": null, "markers": "None", "file": "setup.cfg", "requirement": ">=3.1.20220202110738", "requirement_type": "extras_require:all", "extras": []}, {"name": "pytest", "version": null, "markers": "None", "file": "setup.cfg", "requirement": "<7.2,>=6.0.2", "requirement_type": "extras_require:all", "extras": []}, {"name": "pytest-cov", "version": null, "markers": "None", "file": "setup.cfg", "requirement": null, "requirement_type": "extras_require:all", "extras": []}, {"name": "pytest-xdist", "version": null, "markers": "None", "file": "setup.cfg", "requirement": null, "requirement_type": "extras_require:all", "extras": []}]}'
```

Here's the formatted output:

```js
{
  "result": [
    {
      "name": "cwl-utils",
      "version": null,
      "markers": "None",
      "file": "setup.cfg",
      "requirement": ">=0.12",
      "requirement_type": "install_requires",
      "extras": []
    },
    {
      "name": "miniwdl",
      "version": null,
      "markers": "None",
      "file": "setup.cfg",
      "requirement": null,
      "requirement_type": "install_requires",
      "extras": []
    },
    {
      "name": "regex",
      "version": null,
      "markers": "None",
      "file": "setup.cfg",
      "requirement": null,
      "requirement_type": "install_requires",
      "extras": []
    },
    {
      "name": "mypy",
      "version": "0.941",
      "markers": "None",
      "file": "setup.cfg",
      "requirement": "==0.941",
      "requirement_type": "extras_require:mypy",
      "extras": []
    },
    {
      "name": "types-requests",
      "version": null,
      "markers": "None",
      "file": "setup.cfg",
      "requirement": null,
      "requirement_type": "extras_require:mypy",
      "extras": []
    },
    {
      "name": "types-setuptools",
      "version": null,
      "markers": "None",
      "file": "setup.cfg",
      "requirement": null,
      "requirement_type": "extras_require:mypy",
      "extras": []
    },
    {
      "name": "cwltest",
      "version": null,
      "markers": "None",
      "file": "setup.cfg",
      "requirement": null,
      "requirement_type": "extras_require:test",
      "extras": []
    },
    {
      "name": "cwltool",
      "version": null,
      "markers": "None",
      "file": "setup.cfg",
      "requirement": ">=3.1.20220202110738",
      "requirement_type": "extras_require:test",
      "extras": []
    },
    {
      "name": "pytest",
      "version": null,
      "markers": "None",
      "file": "setup.cfg",
      "requirement": "<7.2,>=6.0.2",
      "requirement_type": "extras_require:test",
      "extras": []
    },
    {
      "name": "pytest-cov",
      "version": null,
      "markers": "None",
      "file": "setup.cfg",
      "requirement": null,
      "requirement_type": "extras_require:test",
      "extras": []
    },
    {
      "name": "pytest-xdist",
      "version": null,
      "markers": "None",
      "file": "setup.cfg",
      "requirement": null,
      "requirement_type": "extras_require:test",
      "extras": []
    },
    {
      "name": "mypy",
      "version": "0.941",
      "markers": "None",
      "file": "setup.cfg",
      "requirement": "==0.941",
      "requirement_type": "extras_require:all",
      "extras": []
    },
    {
      "name": "types-requests",
      "version": null,
      "markers": "None",
      "file": "setup.cfg",
      "requirement": null,
      "requirement_type": "extras_require:all",
      "extras": []
    },
    {
      "name": "types-setuptools",
      "version": null,
      "markers": "None",
      "file": "setup.cfg",
      "requirement": null,
      "requirement_type": "extras_require:all",
      "extras": []
    },
    {
      "name": "cwltest",
      "version": null,
      "markers": "None",
      "file": "setup.cfg",
      "requirement": null,
      "requirement_type": "extras_require:all",
      "extras": []
    },
    {
      "name": "cwltool",
      "version": null,
      "markers": "None",
      "file": "setup.cfg",
      "requirement": ">=3.1.20220202110738",
      "requirement_type": "extras_require:all",
      "extras": []
    },
    {
      "name": "pytest",
      "version": null,
      "markers": "None",
      "file": "setup.cfg",
      "requirement": "<7.2,>=6.0.2",
      "requirement_type": "extras_require:all",
      "extras": []
    },
    {
      "name": "pytest-cov",
      "version": null,
      "markers": "None",
      "file": "setup.cfg",
      "requirement": null,
      "requirement_type": "extras_require:all",
      "extras": []
    },
    {
      "name": "pytest-xdist",
      "version": null,
      "markers": "None",
      "file": "setup.cfg",
      "requirement": null,
      "requirement_type": "extras_require:all",
      "extras": []
    }
  ]
}
```

Which includes the `mypy==0.941` dependency above :tada: So I believe Dependabot will keep sending updates to `mypy` and other groups/extras dependencies.

Bruno

[^1]: https://github.com/dependabot/dependabot-core/blob/main/python/helpers/lib/parser.py

File: setup.py [hunk 2]
```diff
@@ -1,22 +1,5 @@
...
-    url="https://github.com/common-workflow-lab/wdl-cwl-translator",
-    version="0.0.1",
-    author="Dinithi Wickramaratne",
-    author_email="diniwick124@gmail.com",
-    long_description_content_type="text/markdown",
-    packages=find_packages(),
-    setup_requires=[] + pytest_runner,
-    install_requires=["cwl-utils", "miniwdl", "regex"],
-)
+setup()
```
Comment (mr-c) [inline setup.py:5]: a `setup.py` shouldn't be requited anymore with https://packaging.python.org/en/latest/tutorials/packaging-projects/#creating-pyproject-toml

Comment (kinow) [reply setup.py]: Ah! Today I Learned! Updated with a `pyproject.toml`, removing the `setup.py` (we can add it back later if we ever need some Python code in our build.)

Comment (kinow) [reply setup.py]: Updated `Makefile` as well. Not sure if the `pydocstyle` targets are correct though, but they passed locally for me.

File: Makefile [hunk 3]
```diff
@@ -48,7 +47,7 @@ install-dep: install-dependencies
 
 install-dependencies: FORCE
 	pip install --upgrade $(DEVPKGS)
-	pip install -r requirements.txt
+	pip install .[all]
```
[the remark this answers is not in the document]
Comment (kinow) [reply Makefile:51]: Hmmm, true. Is this `install-dependencies` useful for the Debian package build? A `pip install -e .` will install dependencies and create the symlink (which can be removed in another step), e.g.

```bash
pip install -e .[all]
pip uninstall --yes wdl2cwl
```

WDYT?

**Comment by kinow** [comment]
TIL², I believe we can remove the `mypy.ini` file as well, and use `pyproject.toml`: https://mypy.readthedocs.io/en/stable/config_file.html#using-a-pyproject-toml-file

```
[[tool.mypy.overrides]]
module = 'wdl2cwl'
show_error_context = true
show_column_numbers = true
show_error_codes = true
pretty = true
strict = True

[[tool.mypy.overrides]]
module = "ruamel.*"
ignore_errors = True

[[tool.mypy.overrides]]
module = "lark.*"
ignore_missing_imports = True
```

Will try it later today and see what else would need to be updated too.

**Comment by kinow** [comment]
>Will try it later today and see what else would need to be updated too.

Syntax was a little different, but appears to have worked. `make mypy` passing locally for me.

**Pull request merged**